R/track_rust_source.R

```diff
@@ -0,0 +1,80 @@
+# See https://github.com/r-lib/pkgbuild/blob/2aaecf838eb023b788b7a31e267ac644d616dae0/R/compile-dll.R
+
+get_library_path <- function(path = ".") {
+  pkg <- desc::desc(rprojroot::find_package_root_file("DESCRIPTION", path = path))
+  pkg_name <- pkg$get("Package")
+  fs::path(
+    glue::glue(
+      "src",
+      "{pkg_name}{.Platform$dynlib.ext}",
+      .sep = .Platform$file.sep
+    )
+  )
+}
+
+get_rust_files <- function(path = ".") {
+  rust_root <- rprojroot::find_package_root_file("src", "rust", path = path)
```

Comment:
I would either not get into this at all (it'll never be 100% right, since we're not tracking dependencies in the way `cargo` does) or recursively find all `*.rs` and `Cargo.toml` files under `src`.

Reply:
We of course cannot outdo `cargo` or `make` itself. What we can do is track 'obvious' changes, like modifying rust source or `Cargo.toml`. I will remove `rust` subdir from path -- any `*.rs` file under `pkg_root/src/` should be tracked.